### DISCUSSION TITLE
Feature request - Browseraurus for Linux

### DISCUSSION THREAD

**Discussion by bigjdunham** [will-stone/browserosaurus#465] [Ideas]
Browsersaurus is amazing and I use it extensively in MacOS, but I find myself using Linux (specifically Ubuntu) more and more every day, and I can't find a tool similar to Browsersaurus. So... any chance you can develop a version for Linux? I'd be happy to chip in some support cash to help that happen! ;)

**Comment by will-stone** [comment]
Hi @bigjdunham Thanks for using Browserosaurus and the kind words you said about it. Before I continue, I would be interested to know how much you'd be willing to pay? Please answer that truthfully before reading on. Now factor in the following:

- I do not own a PC, and would need to purchase a development PC and install Linux on it. I used to use Linux before moving over to MacBooks about 10 years ago so I could certainly install it and get up and running. But let's say to get to dev-ready this would take me ~ 0.5 days.
- Whilst Electron is cross OS compatible, Browserosaurus uses macOS-specific APIs to find and launch browsers. I would need to investigate what the equivalent of these are in Linux (specifically Ubuntu) and determine how to integrate them with Browserosaurus. Let's call this a day's work ~ 1 day.
- Compiling and releasing Electron apps can be a real pain (it certainly is on macOS), there's always so many gotchas. I would allow another day just for this ~ 1 day.
- Now I'd have to maintain this version of Browserosaurus, whilst it's hard to quote for this, I would add another couple of days to cover the lifetime of the app ~ 2 days.

I searched "average software engineer salary uk" and the value from Glassdoor was £53,392. Putting this number into [The Salary Calculator](https://www.thesalarycalculator.co.uk) gave £205.35 a day. Let's also say £400 for an average dev PC. Therefore my quote would be 4.5 * 205.35 + 400 = £1,324.

I'm really pleased you like Browserosaurus but as you can see, offering cash for features isn't really the done thing in open source, and this is way more than a mere feature request.

In all honesty, _you're_ in the best place to develop this. The code is here, I can point you to the relevant bits, and the most important fact is you already have the machine you'd like to get it working on.

I toy with the idea of moving back to Linux all the time: do I want to be confined by Apple's rules? Do I agree with the direction Apple is going? Can I take Apple breaking the release process... again?! But at the end of the day, it's unlikely. Apple, for their faults, get a _lot_ right. And I can't see me moving away any time soon... but never say _never_... right? 😊

If you would like to make a fork of Browserosaurus and create a Linux-compatible version, the only thing I ask is that you please change the name. I once suggested "Browseratops" to someone wanting to make a Linux version, they never did so it's still available.

**Comment by schneebonus** [comment]
Made a hacky version of the core functionality for linux: https://github.com/schneebonus/Browseropegomastax
Not a fork of Browserosaurus but it gets the job done.

"Browseratops" is still available for a real fork.
